Turning helper methods of annotation_categories_helper into class methods in the AnnotationCategory model

mimischly7 opened the pull request:
<!--- Provide a summary of your changes in the Pull Request Title above. -->
<!--- If this is a work in progress (not yet ready to be merged), make this a draft pull request. -->

## Motivation and Context
<!--- Why is this pull request required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
Some helper methods inside `app/helpers/annotation_categories_helper` really belonged as class ("static") methods inside the AnnotationCategory model. The purpose of this pull request relates to a small refactoring in the codebase required to move these helper methods to the AnnotationCategory model.


## Your Changes
<!--- Describe your changes here. -->
<!--- Include how your changes may affect other areas of the application, if relevant. -->
**Description**: Refactored the methods `annotation_categories_to_yml` and `upload_annotations_from_yaml` from the helper module `app/helpers/annotations_categories_helper` to move them into the AnnotationCategory. For `upload_annotations_from_yaml`, its implementation is using the `current_role` method from the `app/lib/session_handler` module. This method is directly accessible from within controllers and helpers, but not in models. Thus, I added a `current_role` parameter to the `upload_annotations_from_yaml` class method, so its invocations (all of which happen in a model, where the `current_role` is directly accessible) pass a valid `current_role`.


**Type of change** (select all that apply):
<!--- Put an `x` in all the boxes that apply. -->
<!--- Remove any lines that do not apply. -->

- [x] Refactoring (internal change to codebase, without changing functionality)


## Testing
<!--- Please describe in detail how you tested this pull request. -->
<!--- This can include tests you added and manual testing through the web interface. -->
I ensured all `rspec` tests run successfully. In particular one test was failing initially (`spec/controllers/annotation_categories_controller_spec.rb`), so I called `rspec` for just that test until I solved the issue. Then I ran the entire test suite again to ensure everything is fine. Additionally, I checked that the UI is still functioning as expected with regard to the components related to my change. 


## Questions and Comments (if applicable)
<!-- Ask any questions you have for the maintainers of this project regarding this PR. -->
<!-- Please describe the steps you have already taken to find the answer to your question. -->
<!-- This will ensure that we can give you clear and relevant advice. -->
<!-- If you have additional comments add them here as well. -->


## Checklist

- [x] I have performed a self-review of my own code.
- [x] I have verified that the pre-commit.ci checks have passed. <!-- (check after opening pull request) -->
- [x] I have verified that the CI tests have passed. <!-- (check after opening pull request) -->
- [x] I have reviewed the test coverage changes reported on Coveralls. <!-- (check after opening pull request) -->
- [ ] I have added tests for my changes. <!-- (delete this checklist item if not applicable) -->
- [x] I have updated the Changelog.md file. <!-- (delete this checklist item if not applicable) -->
- [ ] I have made changes to the documentation and linked to that pull request below. <!-- (delete this checklist item if not applicable) -->

### Pull request to make documentation changes (if applicable)
<!--- Add a link to a pull request on the MarkUsProject/Wiki repo -->

david-yz-liu reviewed (commented):
@mimischly7 good work! A few overall comments:

1. Now that you've moved the methods, you can check for whether the `include` of `AnnotationCategoriesHelper` can be removed or not.
2. Please add yourself to the list of contributors.
3. For the pull request template, under "type of change" you should *delete* the items that don't apply, so that they don't show up as "incomplete" tasks in the GitHub UI.

mimischly7 commented:
Thank you professor, I will get started with these right away. I had a small question:

Right before I made the pull request last week, I updated my local git repository with upstream (by first pulling from the `upstream master` to my local master, and then merging my local `master` branch into my local `fix-static-methods` branch). Then I ran the `rspec` test suite and there were no failures (so afterwards I confidently made the pull request).
However, I see here in this pull request that a check failed (an image of this is seen below). So I went to my local repository and updated with upstream again to rerun `rspec`, and after all tests ran I got the following output (which I had not gotten again previously): "Pending: (Failures listed here are expected and do not affect your suite's status)". I also included an image with the exact output.

Is this a problem on my side, or is it anticipated? I am thinking something changed in the upstream since I last made the pull request maybe.


<img width="830" alt="Screenshot 2023-09-26 at 12 34 30 PM" src="https://github.com/MarkUsProject/Markus/assets/129094642/df07879e-de8a-4e54-b0a1-11b37d18d745">
<img width="1313" alt="Screenshot 2023-09-26 at 2 02 22 PM" src="https://github.com/MarkUsProject/Markus/assets/129094642/b84a69a3-56e3-4db6-b684-9192c50984b1">

Have a great day!

Mimis

david-yz-liu commented:
@mimischly7 oh don't worry about that, it's a problem with how our tests are currently configured on GitHub Actions. I noticed it while reviewing your and Bruce's PRs, and I've actually just made a fix (#6750)

david-yz-liu commented on app/controllers/assignments_controller.rb:
```diff
@@ -2,7 +2,6 @@ class AssignmentsController < ApplicationController
   include RepositoryHelper
   include RoutingHelper
   include CriteriaHelper
-  include AnnotationCategoriesHelper
```
Okay great. It seems like you can delete the `include` inside the test file as well.

mimischly7 replied:
(done)

david-yz-liu commented on Changelog.md:
```diff
@@ -1,6 +1,7 @@
 # Changelog
 
 ## [unreleased]
+- Convert helper methods of module into class methods of model (#6747)
```
You should be more specific here and describe exactly what modules and models were involved
